lib/tools/apk-utils.js

```diff
@@ -53,19 +53,21 @@ apkUtilsMethods.isAppInstalled = async function isAppInstalled (pkg) {
  *
  * @param {string} uri - The name of URI to start.
  * @param {string} pkg - The name of the package to start the URI with.
+ * @param {boolean} waitForLaunch - Optional parameter of waitForLaunch caps to add wait in adb command
  */
-apkUtilsMethods.startUri = async function startUri (uri, pkg) {
+apkUtilsMethods.startUri = async function startUri (uri, pkg, opts = { waitForLaunch: true }) {
   if (!uri || !pkg) {
     throw new Error('URI and package arguments are required');
   }
 
-  const args = [
-    'am', 'start',
-    '-W',
-    '-a', 'android.intent.action.VIEW',
-    '-d', uri.replace(/&/g, '\\&'),
-    pkg,
-  ];
+  const args = ['am', 'start'];
+  if (opts.waitForLaunch) {
```

Comment:
waitForLaunch could be destructed before

Reply:
@mykola-mokhnach Sorry I am not able to understand it better.

Reply:
have you checked the example above?

Reply:
Ohk, Got it. I am going through it. Thanks for guiding me.

Reply:
How it looks now?

  @param {string} pkg - The name of the package to start the URI with.
  @param {StartUriOptions} opts
apkUtilsMethods.startUri = async function startUri (uri, pkg, opts = {}) {
  if (!uri || !pkg) {
    throw new Error('URI and package arguments are required');
  }

  const {
    waitForLaunch = true,
  } = opts;

  const args = ['am', 'start'];
  if (waitForLaunch) {
    args.push('-W');
  }